server/datastore/mysql/migrations/tables/20230328091816_EnsureUniformCollation.go

```diff
@@ -3,14 +3,170 @@ package tables
 import (
 	"bytes"
 	"database/sql"
+	"encoding/json"
 	"fmt"
 	"text/template"
 
 	"github.com/jmoiron/sqlx"
 )
 
 func init() {
-	MigrationClient.AddMigration(Up_20230315104937, Down_20230315104937)
+	MigrationClient.AddMigration(Up_20230328091816, Down_20230328091816)
+}
+
+func fixupSoftware(tx *sql.Tx, collation string) error {
+	rows, err := tx.Query(fmt.Sprintf(`
+         SELECT
+           COUNT(*) as total,
+           CONCAT('[', GROUP_CONCAT(id SEPARATOR ','), ']') as ids
+         FROM software
+         GROUP BY
+           name      COLLATE %s,
+           "version" COLLATE %s,
+           source    COLLATE %s,
+           "release" COLLATE %s,
+           vendor    COLLATE %s,
+           arch      COLLATE %s
+         HAVING total > 1
+         COLLATE %s`, collation, collation, collation, collation, collation, collation, collation))
+
+	if err != nil {
+		return err
+	}
+
+	defer rows.Close()
+	var idGroups [][]uint
+	for rows.Next() {
```

Comment:
here and below, I tried to use `sqlx.Tx`, but I was getting a panic while it tried to decode the return value. In the end I gave up and resorted to this.